### PR TITLE
Add valid shorthand rules to KB even when not all shorthand rules are valid

### PR DESCRIPTION
**Pull request by gj**
PR checklist:

- [ ] Added changelog entry.

### CODE REVIEW

**Review by gkaemmer — APPROVED** [review]
Looks good. It's a bit scary that this doesn't break any tests -- would it be easy to add a test for the case where _some_ shorthand rules are inserted into the KB despite errors?

**Comment by gj** [comment]
> Looks good. It's a bit scary that this doesn't break any tests -- would it be easy to add a test for the case where _some_ shorthand rules are inserted into the KB despite errors?

Great idea; will do

File: polar-core/src/polar.rs [hunk 1]
```diff
@@ -398,4 +395,65 @@ mod tests {
         );
         assert!(!polar.kb.read().unwrap().has_rules());
     }
+
+    #[test]
+    fn test_valid_shorthand_rules_rewritten_in_presence_of_other_issues() {
+        let polar = Polar::new();
+        let src = r#"
+            allow(actor, action, resource) if
```
Comment (gkaemmer) [inline polar-core/src/polar.rs:403]: Not sure I totally understand this test — what’s the error that would have prevented these rules from being rewritten into the KB?

Comment (gj) [reply polar-core/src/polar.rs:403]: Ah yeah that's a good shout — it's not totally obvious. I'll add a comment explaining it

Comment (gj) [reply polar-core/src/polar.rs:400]: Simpler test in af960f69

**Pull request merged**